graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
```

Comment:
@stevvooe @dmcgowan IIRC, `docker/distribution` wasn't vendored in when I originally wrote `utils.DigestReference`. +1 to removing it, now that we can call `digest.ParseDigest`.